golang/vaas/pkg/authenticator/authenticator.go

```diff
@@ -0,0 +1,62 @@
+package authenticator
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	msg "vaas/pkg/messages"
+)
+
+type ClientCredentialsGrantAuthenticator interface {
+	GetToken(accessToken *string) error
+}
+
+type clientCredentialsGrantAuthenticator struct {
+	cliendId      string
+	clientSecret  string
+	tokenEndpoint string
+	httpClient    *http.Client
+}
+
+func New(clientId string, clientSecret string, tokenEndpoint string) *clientCredentialsGrantAuthenticator {
```

Comment:
If you already have an interface for the returned type, you should use it. Returning unexported types could be quite annoying for users because you can not create a variable of this type and you have to know, that such an interface exists. For example:

```go
// This works 
myAuthenticator := authenticator.New(...)

// but things like this do not 
var myAuthenticator *autheticator.clientCredentialsGrantAuthenticator
```